src/AppBundle/Service/ProductSelectionService.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: hamler
+ * Author: Viktoras Bezubec
+ * Date: 17.11.18
+ * Time: 10.56
+ */
+
+namespace AppBundle\Service;
+
+use AppBundle\Entity\Order;
+use AppBundle\Entity\Product;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Session\Session;
+
+class ProductSelectionService
+{
+    private $em;
+    private $container;
+    private $facebook;
+    private $hoursToRevealSecret;
+
+
+    /**
+     * ProductSelectionService constructor.
+     *
+     * @param EntityManager $em
+     * @param ContainerInterface $container
+     * @param Session $session
+     * @param FacebookInfoService $facebook
+     */
+    public function __construct(EntityManager $em, ContainerInterface $container, FacebookInfoService $facebook)
+    {
+        $this->em = $em;
+        $this->container = $container;
+        $this->facebook =  $facebook;
+        $this->hoursToRevealSecret = $container->getParameter('secret_reveal_time') * 24;
```

Comment:
you can pass needed parameters in services.yml directly to constructor, like:

AppBundle\SomeFolder\SomeService
        arguments:
            - "%your_parameter%"